src/rql/index.ts

```diff
@@ -109,17 +135,30 @@ export function rqlBuilder(rql?: RQLString | string): RQLBuilder {
     },
   };
 
-  function processQuery(operation: string, value: string) {
-    if (value.includes(' ')) {
-      console.log(
-        'A space has been detected while building the rql, please be aware that problems can arise'
-      );
+  function processValue(value: string) {
+    if (!shouldDoubleEncode) {
+      return value;
     }
+
+    const singleEncodedValue = encodeValue(value);
+    const doubleEncodedValue = encodeValue(singleEncodedValue);
+
+    return doubleEncodedValue;
+  }
+
+  function encodeValue(value: string) {
+    return encodeURIComponent(value).replace(
+      /[-_.!~*'()]/g,
+      c => `%${c.charCodeAt(0).toString(16).toUpperCase()}`
+    );
+  }
+
+  function processQuery(operation: string, value: string) {
     returnString = returnString
       .concat(returnString.length > 0 ? '&' : '')
```

Comment:
Annoying question: but it is problematic that we use the shorthand here? Not sure if the & breaks something 🤔

Reply:
Great notice! 🙏

Reply:
Tested with Aäron and this does not negatively effect the outcome thus it will be left as it is! 👍